apollo-normalized-cache-sqlite/src/commonMain/kotlin/com/apollographql/apollo3/cache/normalized/sql/sqlNormalizedCacheFactory.kt

```diff
@@ -0,0 +1,33 @@
+package com.apollographql.apollo3.cache.normalized.sql
+
+import com.apollographql.apollo3.cache.normalized.api.NormalizedCacheFactory
+import com.apollographql.apollo3.cache.internal.blob.BlobDatabase
+import com.apollographql.apollo3.cache.internal.json.JsonDatabase
+import com.apollographql.apollo3.cache.normalized.sql.internal.BlobRecordDatabase
+import com.apollographql.apollo3.cache.normalized.sql.internal.JsonRecordDatabase
+import com.apollographql.apollo3.cache.normalized.sql.internal.RecordDatabase
+import com.squareup.sqldelight.db.SqlDriver
+
+expect class SqlNormalizedCacheFactory internal constructor(
+    driver: SqlDriver,
+    withAge: Boolean = false,
+) : NormalizedCacheFactory
```

Comment:
I'm pretty unhappy about the current state of things of how to build a `SqlNormalizedCacheFactory`. The more I think about it, the more I think the expect/actual here doesn't make much sense. Initialization has to be platform specific so the expect constructor has little use in practice and puts unecessary constraints on the different platforms. Thoughts about deprecating it?

Reply:
I was thinking 

* AndroidSqliteNormalizedCacheFactory
* NativeSqliteNormalizedCacheFactory
* JdbcSqliteNormalizedCacheFactory

Reply:
Oh yeah that works!

Reply:
Nevermind, with app startup, we can expose a common constructor:

```
SqliteNormalizedCacheFactory(name: String?, withDates: Boolean)
```